articles/200_coexistence_and_migration.md

```diff
@@ -0,0 +1,247 @@
+This would also allow building / using / testing ROS 1 with Python 3.
+
+Another example is the mapping of `rosdep` key to the Qt 4 or Qt 5 version of a dependency.
+Package try to avoid changing the rosdep key used for their dependencies but rosdep only considers the platform without the context of the ROS distribution.
+
+**Proposal:** Change request for `rosdep` to consider information beyond the platform to determine the mapping.
+
+**Decision:** With that change the `rosdep` rules should share a single namespace (this is currently already the case).
+
+### Package manifest
+
+In ROS 1 two versions of the package manifest are supported: format 1 specified in [REP 127](http://www.ros.org/reps/rep-0127.html) and format 2 specified in [REP 140](http://www.ros.org/reps/rep-0140.html).
+While ROS 2 only supports format 2 there is no technical limitation behind that design decision.
+It simply reduces the effort to distinguish the different semantics of some depend tags in various parts of the software.
+
+It might be the case that a to-be-specified format 3 will be created to add support for the desired features to describe [dependency groups]() as well as declaring an [ABI version]().
+
+#### ROS dependencies
+
+While a ROS 1 package might depend on `roscpp` and `message_generation` a ROS 2 package depends on `rclcpp` and `rosidl_message_generation` instead.
+
+#### Build type
+
+Almost all build types for ROS 1 and ROS 2 are disjoint except the build type `cmake` which can be processed by both build tools.
+Since it is not desired to change existing `catkin` / `ament_cmake` packages to use the build type `cmake` those are currently different for each ROS version.
+
+### CMake version
+
+ROS 2 requires CMake version 3.5+.
+It should be easily possible to update existing CMake code to be compatible with various versions of CMake.
+
+### Python version
+
+ROS 2 requires Python version 3.5+.
+It should be easily possible to update existing Python code to be compatible with Python 2.7 as well as Python 3.5+.
+
+### CMake API
+
+ROS 1 packages use the CMake API provided by [catkin](https://github.com/ros/catkin).
+ROS 2 packages use a slightly different CMake API provided by [ament_cmake](https://github.com/ament/ament_cmake).
+The rational for the changes and the different name are described in [this article](http://design.ros2.org/articles/ament.html).
+
+**Idea:** Create a ROS 2 package names `catkin` which provides the same CMake API as the ROS 1 package (minus unavailable features like the devel space) by mapping the calls to `ament_cmake` API.
+In addition it is expected that existing packages need to call a new function at the end of their CMake code to account for the fact that `ament_package` needs to be invoked *after* all targets where `catkin_package` needs to be invoked *before* them.
+
+### Message / service definitions
```

Comment:
maybe should be level 2 header?

Reply:
The intention was to have this subsection under `Technical Constraints`